### ISSUE TITLE
Error when importing 

### ISSUE THREAD

**Issue by danifranco** [bioimage-io/core-bioimage-io-python#369]
Hi,

There is a problem when running a code that imports bioimage.core inside docker (I'm using ``bioimageio.core==0.5.9``). The problem is that you rely on ``getpass.getuser()`` to create the temporal directory for storing bioimage.core cache, which can throw an error if the user inside docker does not exist, which is the case when you run as non-root user. I can't create the specific user in the Dockerfile as the container is going to be public and many users with different names will be using it. This error is common in other projects too: https://github.com/mesonbuild/meson/issues/11662 and https://github.com/Azure/azure-cli/issues/2100 . 

I suggest to add a try/catch in the call of ``getpass.getuser()`` and create an alternative cache path if the username is not available. I tried also disabling the cache with ``os.environ["BIOIMAGEIO_USE_CACHE"] = "no"`` but still the error is thrown. Maybe I need something else to disable completely the cache, not sure about this. 

<details>
  <summary>The complete error is here in case you are insterested:</summary>

```
Traceback (most recent call last):
  File "/installations/BiaPy/utils/misc.py", line 17, in <module>
    from bioimageio.core.build_spec import build_model
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/core/__init__.py", line 6, in <module>
    from .resource_io import (
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/core/resource_io/__init__.py", line 1, in <module>
    import bioimageio.spec
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/__init__.py", line 1, in <module>
    from . import collection, model, rdf, shared
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/collection/__init__.py", line 1, in <module>
    from . import v0_2
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/collection/v0_2/__init__.py", line 1, in <module>
    from . import converters, raw_nodes, schema, utils
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/collection/v0_2/converters.py", line 4, in <module>
    from bioimageio.spec.rdf.v0_2.converters import maybe_convert as maybe_convert_rdf
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/rdf/__init__.py", line 1, in <module>
    from . import v0_2
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/rdf/v0_2/__init__.py", line 1, in <module>
    from . import converters, raw_nodes, schema, utils
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/rdf/v0_2/raw_nodes.py", line 19, in <module>
    from bioimageio.spec.shared.raw_nodes import RawNode, ResourceDescription, URI
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/shared/__init__.py", line 6, in <module>
    from ._resolve_source import (
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/shared/_resolve_source.py", line 17, in <module>
    from .common import (
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/site-packages/bioimageio/spec/shared/common.py", line 63, in <module>
    os.getenv("BIOIMAGEIO_CACHE_PATH", pathlib.Path(tempfile.gettempdir()) / getpass.getuser() / "bioimageio_cache")
  File "/installations/miniconda3/envs/BiaPy_env/lib/python3.8/getpass.py", line 169, in getuser
    return pwd.getpwuid(os.getuid())[0]
KeyError: 'getpwuid(): uid not found: 1001'
```
</details>

Thank you so much!

**Comment by FynnBe** [comment]
Hi @danifranco  , thank you for reporting this issue!

We will soon (in a couple weeks) release bioimageio.core 0.6, which will use [pooch](https://www.fatiando.org/pooch/latest/) for cache management and thus won't have this issue anymore. 
Please let me know if you need a fix asap and cannot wait for 0.6.

**Comment by danifranco** [comment]
Don't worry, I'm okay waiting until 0.6 version. Thank you so much!
